Sources/Apodini/Request/Dynamics.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  Dynamics.swift
+//  
+//
+//  Created by Max Obermeier on 10.12.20.
+//
+
+import Foundation
+
+/// Dynamics provides same functionality as `DynamicProperty`, but for elements that are only known
+/// at statup and not compile-time. That is, `Dynamics` stores named elements and makes them
+/// discoverable to the Apodini runtime framework.  This can be used to e.g. delay the decision, which
+/// `Parameter`s are exported to startup-time
+@dynamicMemberLookup
+public struct Dynamics<Element> {
```

Comment:
I think it would result in a nice interface if `Dynamics` or `Injectables` would also work as a property wrapper as shown in the slides of @hendesi in the intermediate presentation: https://confluence.ase.in.tum.de/display/SERVERSIDESWIFT2021/Intermediate+Presentation?preview=/66506934/66513049/Intermediate_Presentation_Felix_Desiderato.pdf
I think the syntax would offer a very nice abstraction and the developer using Apodini could simply work on a dictionary of the type `[String: Injectable]` or `[String: RequestInjectable]` or even a more concrete type if the developer e.g. only uses `@Parameter`s and could therefore use a dictionary of that type `[String: Parameter]`.

Reply:
I made `Dynamics` an `@propertyWrapper` with get-only `wrappedValue` that exposes the `Dynamics`' `elements`.

> the developer using Apodini could simply work on a dictionary of the type [String: Injectable] or [String: RequestInjectable]

I do not get why you think that would be useful or even possible. `RequestInjectable` is an internal protocol and provides no interface to the user of Apodini. I also don't get what a user of Apodini should do with an `inject` function on those elements if the more abstracted `Injectable` protocol were to be made public.

Reply:
Thanks a lot, I think this offers a nice API as already mentioned in my comment in the thread below (https://github.com/Apodini/Apodini/pull/56/files#r546222581)!
I tried to showcase why this would be useful and what the reason behind the `Injectable` protocol is in the comment thread above (https://github.com/Apodini/Apodini/pull/56/files#r546220357) and the initial intention behind `Injectable` and the motivation why I don't think that there is a need for a public `inject(...)` method in the thread above (https://github.com/Apodini/Apodini/pull/56/files#r546223303). I hope this clears up my motivation on why I would suggest to use an `[String: Injectable]` dictionary in `Injectables`/`Dynamics`. I agree with you that constraining it to `[String: RequestInjectable]` would not be useful as `RequestInjectable` is and should stay internal.